openedx/features/enterprise_support/api.py

```diff
@@ -569,6 +570,44 @@ def get_enterprise_learner_data(user):
             return enterprise_learner_data['results']
 
 
+@enterprise_is_enabled()
+def get_enterprise_learner_portal_enabled_message(request):
+    """
+    Returns True if the user is linked to an Enterprise with the Learner Portal enabled.
+
+    Note: request.session['enterprise_customer'] will be used in case the user is linked to
+        multiple Enterprises. Otherwise, it won't exist and the Enterprise Learner data
+        will be used. If that doesn't exist return None.
+
+    Args:
+        request: request made to the LMs dashboard
+    """
+    learner_data = get_enterprise_learner_data(request.user)
```

Comment:
if `learner_data` isn't used when `request.session['enterprise_customer']` exists, should we move this line to be within the `elif` conditional on line 588? That way, we don't need to make an unnecessary API request.

Reply:
Yeah I wrestled with this for a bit myself. If you take a look at the original commit you'll see that's what I had. However, when done that way the same API call is made twice so I assigned the value of `get_enterprise_learner_data` to a variable beforehand to avoid that. In another alternative solution: if I make the elif a separate if statement to assign a value to `learner_data` after the initial if statement and then check the value, I'd be allowing `enterprise_customer` to be overwritten

Reply:
solution discussed in slack, fix coming soon